packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,364 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import React from "react";
+
+import {
+	Button,
+	Dropdown,
+	DropdownProps,
+	Input,
+	InputOnChangeData,
+	Option,
+	makeStyles,
+	shorthands,
+	tokens,
+} from "@fluentui/react-components";
+import {
+	DataEdit,
+	EditData,
+	EditType,
+	FluidObjectValueNode,
+} from "@fluid-experimental/devtools-core";
+
+import { useMessageRelay } from "../../MessageRelayContext";
+import { TreeHeader } from "./TreeHeader";
+
+/**
+ * Input to {@link EditableView}
+ */
+export interface EditableViewProps {
+	node: FluidObjectValueNode;
+	containerKey: string;
+	label: string;
+}
+
+/**
+ * Input to EditableComponents
+ */
+interface EditableComponentProps {
+	node: FluidObjectValueNode;
+	editProp: EditState | undefined;
```

Comment:
Nit: I'm not sure "Prop" adds anything to the variable names here. What do you think of something like "activeEdit" and "updateEdit". In my mind "active" indicates that the edit is some ongoing thing.

Reply:
We should also document these, and be sure to note what it means for `editProp` to be undefined (no ongoing edit), and for the set function to be passed undefined (cancel edit).